components/script/canvas_state.rs

```diff
@@ -1892,6 +1951,19 @@ impl CanvasState {
     }
 }
 
+pub(crate) fn is_supported_canvas_size(size: Size2D<u64>) -> bool {
+    if size.is_empty() {
+        return false;
+    }
+    if size.width > MAX_CANVAS_SIZE || size.height > MAX_CANVAS_SIZE {
+        return false;
+    }
+    if size.area() > MAX_CANVAS_AREA {
+        return false;
+    }
+    true
```

Comment:
I would just combine this one:
```suggestion
    !size.is_empty() &&
        size <= Size2D::new(MAX_CANVAS_SIZE, MAX_CANVAS_SIZE) &&
        size.area() < MAX_CANVAS_AREA
```

Reply:
And `size <= Size2D::new(MAX_CANVAS_SIZE, MAX_CANVAS_SIZE)` is not supported
(PartialOrd is not implemeted for Size2D), so I replace it to 
`size.greater_than(Size2D::new(MAX_CANVAS_SIZE, MAX_CANVAS_SIZE)).none()`